lisa/features/gpu.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+from enum import Enum
+from typing import Any
+
+from lisa.base_tools.wget import Wget
+from lisa.feature import Feature
+from lisa.operating_system import Redhat, Ubuntu
+from lisa.tools import Uname
+from lisa.util import LisaException, SkippedException
+from lisa.util.logger import get_logger
+
+FEATURE_NAME_GPU = "Gpu"
+
+ComputeSDK = Enum(
+    "ComputeSDK",
+    [
+        # GRID Driver
+        "GRID",
+        # CUDA Driver
+        "CUDA",
+    ],
+)
+
+# Link to the latest GRID driver
+# The DIR link is
+# https://download.microsoft.com/download/9/5/c/95c667ff-ab95-4c56-89e0-e13e9a76782d/NVIDIA-Linux-x86_64-460.32.03-grid-azure.run
+GRID_DRIVER = "https://go.microsoft.com/fwlink/?linkid=874272"
+
+
+class Gpu(Feature):
+    def __init__(self, node: Any, platform: Any) -> None:
+        super().__init__(node, platform)
+        self._log = get_logger("gpu", self.name(), self._node.log)
+
+    @classmethod
+    def name(cls) -> str:
+        return FEATURE_NAME_GPU
+
+    def _is_supported(self) -> bool:
+        raise NotImplementedError()
+
+    # download and install NVIDIA grid driver
+    def _install_grid_driver(self, driver_url: str = f"{GRID_DRIVER}") -> None:
+        self._log.debug("Starting GRID driver installation")
+        # download and install the NVIDIA GRID driver
+        wget_tool = self._node.tools[Wget]
+        grid_file_path = wget_tool.get(
+            f"{driver_url}",
+            str(self._node.working_path),
+            "NVIDIA-Linux-x86_64-grid.run",
+            executable=True,
+        )
+        result = self._node.execute(
+            f"{grid_file_path} --no-nouveau-check --silent --no-cc-version-check"
+        )
+        if result.exit_code != 0:
+            raise LisaException(
+                "Failed to install the GRID driver! "
+                f"exit-code: {result.exit_code} stderr: {result.stderr}"
+            )
+        else:
```

Comment:
Because previous is an safe guard, which throws an exception. This line doesn't need to be in `else` block. It improves the readability a little.

Reply:
Done